adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -171,16 +174,17 @@ func (h *hnsw) CleanUpTombstonedNodes() error {
 }
 
 func (h *hnsw) reassignNeighborsOf(deleteList helpers.AllowList) error {
-	h.Lock()
+	h.RLock()
 	size := len(h.nodes)
-	currentEntrypoint := h.entryPointID
-	h.Unlock()
+	h.RUnlock()
 
```

Comment:
len(h.nodes) might change in the meantime which can lead to a panic in the for loop.
I am not sure if the context does prevent this from happening

Reply:
you are right. the following PR will be focused on solving the panic issues